module/Database/src/Model/CheckoutSession.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Database\Model;
+
+use DateTime;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\GeneratedValue;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\JoinColumn;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\OneToOne;
+use InvalidArgumentException;
+
+use function in_array;
+use function sprintf;
+
+/**
+ * Saved query model.
+ */
+#[Entity]
+class CheckoutSession
+{
+    public const CREATED = 0;
+    public const CANCELLED = 1;
+    public const EXPIRED = 2;
+    public const PENDING = 3;
+    public const FAILED = 4;
+    public const PAID = 5;
```

Comment:
Potentially make this an enum. Then the massive check in `setState()` can be removed.

Reply:
Changed this, as this makes it easier to display information in UI.